command/keyring/keyring_test.go

```diff
@@ -164,3 +189,33 @@ func removeKey(t *testing.T, addr string, key string) {
 		t.Fatalf("bad: %d. %#v", code, ui.ErrorWriter.String())
 	}
 }
+
+func TestKeyringCommand_poolName(t *testing.T) {
+	require.Equal(t, "dc1 (LAN):\n", poolName("dc1", false, ""))
+	require.Equal(t, "dc1 (LAN) [segment1]:\n", poolName("dc1", false, "segment1"))
+	require.Equal(t, "WAN:\n", poolName("dc1", true, ""))
+}
+
+func TestKeyringCommand_formatKeys(t *testing.T) {
+	require.Equal(t, "", formatKeys(map[string]int{}, 0))
+	keys := formatKeys(map[string]int{"key1": 1, "key2": 2}, 2)
+	require.Contains(t, keys, "  key1 [1/2]\n")
+	require.Contains(t, keys, "  key2 [2/2]\n")
+}
+
+func TestKeyringCommand_formatMessages(t *testing.T) {
+	require.Equal(t, "", formatMessages(map[string]string{}))
+	messages := formatMessages(map[string]string{"n1": "hello", "n2": "world"})
+	require.Contains(t, messages, "  ===> n1: hello\n")
+	require.Contains(t, messages, "  ===> n2: world\n")
```

Comment:
Could these use `require.Equal` for stricter assertions? (same above)

Reply:
The problem is the order in which the map is gone through which is random.